### ISSUE TITLE
[Inline assembly] Error on usage of `pc()` with Yul optimizer enabled

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#8852]
## Description

We error on running the Yul optimizer on

```
pragma solidity >= 0.0.0;
contract C {
        function test() public returns (int)
        {
                int8 v0 = int8(0xdc4c7fbf8ec49ab53bc225a482aad76de9d21c98cbf2a163c351192dfaec2e05);
                assembly {
                        v0 := msize()
                }
                return int(v0);
        }
}
```

because

`The msize instruction cannot be used when the Yul optimizer is activated because it can change its semantics. Either disable the Yul optimizer or do not use the instruction.`

We should probably do something similar for `pc()`. Not that it changes optimizer semantics but that one should not rely upon the value of `pc()`

**Comment by axic** [comment]
Should be also "fixed" by #7965

**Comment by Moony6789** [comment]
Hello, bshastry
So how did you fixed this problem. could you explain clearly? 
![image](https://github.com/ethereum/solidity/assets/126175765/0832e24f-8547-4f5d-a696-a96a5c71092e)
![image](https://github.com/ethereum/solidity/assets/126175765/1705d7e8-a0c2-4752-bbd1-b13c5afe115c)
